dataactbroker/helpers/fabs_derivations_helper.py

```diff
@@ -196,30 +190,46 @@ def derive_ppop_location_data(obj, sess, ppop_code, ppop_state_code, county_dict
             obj['place_of_perform_county_co'] = city_info.county_number
             obj['place_of_perform_county_na'] = city_info.county_name
 
-            if (obj['place_of_performance_zip4a'] and obj['place_of_performance_zip4a'].upper() == 'CITY-WIDE'):
-                obj['place_of_performance_scope'] = "City-wide"
-            else:
-                obj['place_of_performance_scope'] = None
-        elif re.match('^[A-Z]{2}\d{4}R$', ppop_code):
-            if (obj['place_of_performance_zip4a'] and obj['place_of_performance_zip4a'].upper() == 'CITY-WIDE'):
-                obj['place_of_performance_scope'] = "City-wide"
-            else:
-                obj['place_of_performance_scope'] = None
-        elif re.match('^[A-Z]{2}\*{5}$', ppop_code):
-            obj['place_of_performance_scope'] = "State-wide"
-        elif re.match('^00\*{5}$', ppop_code):
-            obj['place_of_performance_scope'] = "Multi-state"
-        elif ppop_code == '00FORGN':
-            obj['place_of_performance_scope'] = "Foreign"
-        else:
-            obj['place_of_performance_scope'] = None
-
     # if there's no ppop code, just set them all to None
     else:
         obj['place_of_perform_county_co'] = None
         obj['place_of_perform_county_na'] = None
         obj['place_of_performance_city'] = None
-        obj['place_of_performance_scope'] = None
+
+
+def derive_ppop_scope(obj):
+    """ Deriving place of performance scope values from zip4 and place of performance code
+
+        Args:
+            obj: a dictionary containing the details we need to derive from and to
+    """
+    derived_ppop_scope = None
+    ppop_code = obj['place_of_performance_code'].upper() if obj['place_of_performance_code'] else None
```

Comment:
Is there a reason we're not just passing the one we already get?

Reply:
Trying to make this function more adaptable, meaning accounting for cases where we can't assume we've upper'd it beforehand.

Reply:
What cases are these?

Reply:
If someone were to call the function or if someone were to copy said function without knowing they were upper'd beforehand.